numpy/__init__.py

```diff
@@ -529,8 +529,11 @@ def hugepage_setup():
     _core.multiarray._multiarray_umath._reload_guard()
 
     # TODO: Remove the environment variable entirely now that it is "weak"
-    _core._set_promotion_state(
-        os.environ.get("NPY_PROMOTION_STATE", "weak"))
+    if (os.environ.get("NPY_PROMOTION_STATE", "weak") != "weak"):
```

Comment:
Assign "weak" to a variable like DEFAULT_PROMOTION_STATE for better maintainability and readability.